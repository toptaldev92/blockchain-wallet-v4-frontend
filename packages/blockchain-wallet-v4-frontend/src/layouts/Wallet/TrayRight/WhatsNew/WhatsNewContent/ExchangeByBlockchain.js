import React from 'react'
import styled from 'styled-components'
import { FormattedMessage } from 'react-intl'
import { connect } from 'react-redux'
import { LinkContainer } from 'react-router-bootstrap'

import { actions, model } from 'data'
import { Button, Text } from 'blockchain-info-components'

const Container = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: flex-start;
  width: 100%;
  height: auto;
`
const Row = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: flex-start;
  width: 100%;
  margin-bottom: 15px;
  margin-bottom: ${props => props.marginBottom};

  & > :not(:last-child) {
    margin-right: 10px;
  }
`
const DarkText = styled(Text).attrs({
  color: 'gray-5',
  size: '16px',
  weight: 300
})`
  display: inline;
  @media (max-width: 1200px) {
    display: ${props => (props.hideOnMobile ? 'none' : 'inline')};
  }
`
const PrimaryText = styled(Text).attrs({
  color: 'brand-primary',
  size: '16px'
})`
  display: inline;
  white-space: nowrap;
`
const GetStartedButton = styled(Button).attrs({
  nature: 'primary',
  fullwidth: true
})`
  font-weight: 500;
  @media (max-width: 1200px) {
    width: 100%;
  }
`

export const ExchangeByBlockchain = ({ kycNotFinished, verifyIdentity }) => (
  <Container>
    <Row marginBottom='24px'>
      <Text color='brand-primary' size='24px' weight={600}>
        <FormattedMessage
          defaultMessage="We've Improved Your Exchange"
          id='layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.improved'
        />
      </Text>
    </Row>
    <Row marginBottom='24px'>
      <DarkText size='14px'>
        <FormattedMessage
          defaultMessage='November 2019'
          id='layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.november2019'
        />
      </DarkText>
    </Row>
    <Row marginBottom='24px'>
      <DarkText>
        <FormattedMessage
          defaultMessage='The faster, smarter way to trade your crypto.'
          id='layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.introducing'
        />
      </DarkText>
    </Row>
    <Row marginBottom='8px'>
      <DarkText>
        <FormattedMessage
          defaultMessage='Upgrade now to enjoy the following benefits:'
          id='layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.upgrade'
        />
      </DarkText>
    </Row>
    <Row marginBottom='8px'>
      <DarkText>
        <PrimaryText>
          <FormattedMessage
            defaultMessage='Lower cost'
            id='layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.lowercost'
          />
        </PrimaryText>
        <DarkText hideOnMobile>{' - '}</DarkText>
        <FormattedMessage
          defaultMessage='Super competitive crypto exchange prices'
          id='layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.lowercost_description'
        />
      </DarkText>
    </Row>
    <Row marginBottom='8px'>
      <DarkText>
        <PrimaryText>
          <FormattedMessage
            defaultMessage='Live rates'
            id='layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.liverates'
          />
        </PrimaryText>
        <DarkText hideOnMobile>{' - '}</DarkText>
        <FormattedMessage
          defaultMessage='You always get the most up to date price'
          id='layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.liverates_Description'
        />
      </DarkText>
    </Row>
    <Row marginBottom='32px'>
      <DarkText>
        <PrimaryText>
          <FormattedMessage
            defaultMessage='Higher limits'
            id='layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.higherlimits'
          />
        </PrimaryText>
        <DarkText hideOnMobile>{' - '}</DarkText>
        <FormattedMessage
          defaultMessage='Limits from $1,000-$25,000'
          id='layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.higherlimits_description'
        />
      </DarkText>
    </Row>
    <Row>
      {kycNotFinished && (
        <GetStartedButton onClick={verifyIdentity}>
          <FormattedMessage
            id='layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.started'
            defaultMessage='Get Started'
          />
        </GetStartedButton>
      )}
      {!kycNotFinished && (
        <LinkContainer to='/swap'>
          <GetStartedButton>
            <FormattedMessage
              id='layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.makeswap'
              defaultMessage='Make Swap'
            />
          </GetStartedButton>
        </LinkContainer>
      )}
    </Row>
  </Container>
)

const mapDispatchToProps = dispatch => ({
  verifyIdentity: () =>
    dispatch(
      actions.components.identityVerification.verifyIdentity(
        model.profile.TIERS[2]
      )
    )
})

export default connect(
  undefined,
  mapDispatchToProps
)(ExchangeByBlockchain)
