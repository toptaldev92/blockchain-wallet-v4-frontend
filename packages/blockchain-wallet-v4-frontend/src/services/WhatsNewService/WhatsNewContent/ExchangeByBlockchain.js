import React from 'react'
import styled from 'styled-components'
import { FormattedMessage } from 'react-intl'
import { connect } from 'react-redux'
import { LinkContainer } from 'react-router-bootstrap'
import { actions, model, selectors } from 'data'
import { Button, Text } from 'blockchain-info-components'
import { Container, Row } from 'components/WhatsNew'
import { equals } from 'ramda'
import media from 'services/ResponsiveService'
const { NONE } = model.profile.KYC_STATES

const DarkText = styled(Text).attrs({
  color: 'gray-5',
  size: '16px',
  weight: 300
})`
  display: inline;
  ${media.laptop`
    display: ${props => (props.hideOnMobile ? 'none' : 'inline')};
  `}
`
const PrimaryText = styled(Text).attrs({
  color: 'brand-primary',
  size: '16px'
})`
  display: inline;
  white-space: nowrap;
`
const GetStartedButton = styled(Button).attrs({
  nature: 'primary',
  fullwidth: true
})`
  font-weight: 500;
  ${media.laptop`
    width: 100%;
  `}
`

export const ExchangeByBlockchain = ({ kycNotFinished, verifyIdentity }) => (
  <Container>
    <Row marginBottom='10px'>
      <Text color='brand-primary' size='24px' weight={600}>
        <FormattedMessage
          defaultMessage="We've Improved Your Exchange"
          id='layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.improved'
        />
      </Text>
    </Row>
    <Row marginBottom='24px'>
      <DarkText size='14px'>
        <FormattedMessage
          defaultMessage='November 2018'
          id='layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.november2018'
        />
      </DarkText>
    </Row>
    <Row marginBottom='24px'>
      <DarkText>
        <FormattedMessage
          defaultMessage='The faster, smarter way to trade your crypto.'
          id='layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.introducing'
        />
      </DarkText>
    </Row>
    <Row marginBottom='8px'>
      <DarkText>
        <FormattedMessage
          defaultMessage='Upgrade now to enjoy the following benefits:'
          id='layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.upgrade'
        />
      </DarkText>
    </Row>
    <Row marginBottom='8px'>
      <DarkText>
        <PrimaryText>
          <FormattedMessage
            defaultMessage='Lower cost'
            id='layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.lowercost'
          />
        </PrimaryText>
        <DarkText hideOnMobile>{' - '}</DarkText>
        <FormattedMessage
          defaultMessage='Super competitive crypto exchange prices'
          id='layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.lowercost_description'
        />
      </DarkText>
    </Row>
    <Row marginBottom='8px'>
      <DarkText>
        <PrimaryText>
          <FormattedMessage
            defaultMessage='Live rates'
            id='layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.liverates'
          />
        </PrimaryText>
        <DarkText hideOnMobile>{' - '}</DarkText>
        <FormattedMessage
          defaultMessage='You always get the most up to date price'
          id='layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.liverates_Description'
        />
      </DarkText>
    </Row>
    <Row marginBottom='32px'>
      <DarkText>
        <PrimaryText>
          <FormattedMessage
            defaultMessage='Higher limits'
            id='layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.higherlimits'
          />
        </PrimaryText>
        <DarkText hideOnMobile>{' - '}</DarkText>
        <FormattedMessage
          defaultMessage='Limits from $1,000-$25,000'
          id='layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.higherlimits_description'
        />
      </DarkText>
    </Row>
    <Row>
      {kycNotFinished && (
        <GetStartedButton onClick={verifyIdentity}>
          <FormattedMessage
            id='layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.started'
            defaultMessage='Get Started'
          />
        </GetStartedButton>
      )}
      {!kycNotFinished && (
        <LinkContainer to='/swap'>
          <GetStartedButton>
            <FormattedMessage
              id='layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.makeswap'
              defaultMessage='Make Swap'
            />
          </GetStartedButton>
        </LinkContainer>
      )}
    </Row>
  </Container>
)

const mapStateToProps = state => ({
  kycNotFinished: selectors.modules.profile
    .getUserKYCState(state)
    .map(equals(NONE))
    .getOrElse(false)
})

const mapDispatchToProps = dispatch => ({
  verifyIdentity: () =>
    dispatch(
      actions.components.identityVerification.verifyIdentity(
        model.profile.TIERS[2]
      )
    )
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ExchangeByBlockchain)
