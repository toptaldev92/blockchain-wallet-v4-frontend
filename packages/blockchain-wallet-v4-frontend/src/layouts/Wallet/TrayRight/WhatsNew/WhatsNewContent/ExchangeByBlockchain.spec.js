import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { ExchangeByBlockchain } from './ExchangeByBlockchain'

jest.mock('blockchain-info-components', () => ({
  Button: 'button',
  Text: 'text'
}))
jest.mock('data', () => ({
  actions: {
    components: {
      identityVerification: {
        verifyIdentity: jest.fn()
      }
    }
  }
}))

describe('ExchangeByBlockchain', () => {
  it('renders correctly with button', () => {
    const component = shallow(<ExchangeByBlockchain kycNotFinished />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with exchange link', () => {
    const component = shallow(<ExchangeByBlockchain />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
