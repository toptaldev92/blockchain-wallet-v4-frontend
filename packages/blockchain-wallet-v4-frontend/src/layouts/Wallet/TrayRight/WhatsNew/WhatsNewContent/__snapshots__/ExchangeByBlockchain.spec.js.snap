// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExchangeByBlockchain renders correctly with button 1`] = `
<ExchangeByBlockchain__Container>
  <ExchangeByBlockchain__Row
    marginBottom="24px"
  >
    <text
      color="brand-primary"
      size="24px"
      weight={600}
    >
      <FormattedMessage
        defaultMessage="We've Improved Your Exchange"
        id="layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.improved"
        values={Object {}}
      />
    </text>
  </ExchangeByBlockchain__Row>
  <ExchangeByBlockchain__Row
    marginBottom="24px"
  >
    <ExchangeByBlockchain__DarkText
      size="14px"
    >
      <FormattedMessage
        defaultMessage="November 2019"
        id="layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.november2019"
        values={Object {}}
      />
    </ExchangeByBlockchain__DarkText>
  </ExchangeByBlockchain__Row>
  <ExchangeByBlockchain__Row
    marginBottom="24px"
  >
    <ExchangeByBlockchain__DarkText>
      <FormattedMessage
        defaultMessage="The faster, smarter way to trade your crypto."
        id="layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.introducing"
        values={Object {}}
      />
    </ExchangeByBlockchain__DarkText>
  </ExchangeByBlockchain__Row>
  <ExchangeByBlockchain__Row
    marginBottom="8px"
  >
    <ExchangeByBlockchain__DarkText>
      <FormattedMessage
        defaultMessage="Upgrade now to enjoy the following benefits:"
        id="layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.upgrade"
        values={Object {}}
      />
    </ExchangeByBlockchain__DarkText>
  </ExchangeByBlockchain__Row>
  <ExchangeByBlockchain__Row
    marginBottom="8px"
  >
    <ExchangeByBlockchain__DarkText>
      <ExchangeByBlockchain__PrimaryText>
        <FormattedMessage
          defaultMessage="Lower cost"
          id="layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.lowercost"
          values={Object {}}
        />
      </ExchangeByBlockchain__PrimaryText>
      <ExchangeByBlockchain__DarkText
        hideOnMobile={true}
      >
         - 
      </ExchangeByBlockchain__DarkText>
      <FormattedMessage
        defaultMessage="Super competitive crypto exchange prices"
        id="layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.lowercost_description"
        values={Object {}}
      />
    </ExchangeByBlockchain__DarkText>
  </ExchangeByBlockchain__Row>
  <ExchangeByBlockchain__Row
    marginBottom="8px"
  >
    <ExchangeByBlockchain__DarkText>
      <ExchangeByBlockchain__PrimaryText>
        <FormattedMessage
          defaultMessage="Live rates"
          id="layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.liverates"
          values={Object {}}
        />
      </ExchangeByBlockchain__PrimaryText>
      <ExchangeByBlockchain__DarkText
        hideOnMobile={true}
      >
         - 
      </ExchangeByBlockchain__DarkText>
      <FormattedMessage
        defaultMessage="You always get the most up to date price"
        id="layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.liverates_Description"
        values={Object {}}
      />
    </ExchangeByBlockchain__DarkText>
  </ExchangeByBlockchain__Row>
  <ExchangeByBlockchain__Row
    marginBottom="32px"
  >
    <ExchangeByBlockchain__DarkText>
      <ExchangeByBlockchain__PrimaryText>
        <FormattedMessage
          defaultMessage="Higher limits"
          id="layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.higherlimits"
          values={Object {}}
        />
      </ExchangeByBlockchain__PrimaryText>
      <ExchangeByBlockchain__DarkText
        hideOnMobile={true}
      >
         - 
      </ExchangeByBlockchain__DarkText>
      <FormattedMessage
        defaultMessage="Limits from $1,000-$25,000"
        id="layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.higherlimits_description"
        values={Object {}}
      />
    </ExchangeByBlockchain__DarkText>
  </ExchangeByBlockchain__Row>
  <ExchangeByBlockchain__Row>
    <ExchangeByBlockchain__GetStartedButton>
      <FormattedMessage
        defaultMessage="Get Started"
        id="layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.started"
        values={Object {}}
      />
    </ExchangeByBlockchain__GetStartedButton>
  </ExchangeByBlockchain__Row>
</ExchangeByBlockchain__Container>
`;

exports[`ExchangeByBlockchain renders correctly with exchange link 1`] = `
<ExchangeByBlockchain__Container>
  <ExchangeByBlockchain__Row
    marginBottom="24px"
  >
    <text
      color="brand-primary"
      size="24px"
      weight={600}
    >
      <FormattedMessage
        defaultMessage="We've Improved Your Exchange"
        id="layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.improved"
        values={Object {}}
      />
    </text>
  </ExchangeByBlockchain__Row>
  <ExchangeByBlockchain__Row
    marginBottom="24px"
  >
    <ExchangeByBlockchain__DarkText
      size="14px"
    >
      <FormattedMessage
        defaultMessage="November 2019"
        id="layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.november2019"
        values={Object {}}
      />
    </ExchangeByBlockchain__DarkText>
  </ExchangeByBlockchain__Row>
  <ExchangeByBlockchain__Row
    marginBottom="24px"
  >
    <ExchangeByBlockchain__DarkText>
      <FormattedMessage
        defaultMessage="The faster, smarter way to trade your crypto."
        id="layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.introducing"
        values={Object {}}
      />
    </ExchangeByBlockchain__DarkText>
  </ExchangeByBlockchain__Row>
  <ExchangeByBlockchain__Row
    marginBottom="8px"
  >
    <ExchangeByBlockchain__DarkText>
      <FormattedMessage
        defaultMessage="Upgrade now to enjoy the following benefits:"
        id="layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.upgrade"
        values={Object {}}
      />
    </ExchangeByBlockchain__DarkText>
  </ExchangeByBlockchain__Row>
  <ExchangeByBlockchain__Row
    marginBottom="8px"
  >
    <ExchangeByBlockchain__DarkText>
      <ExchangeByBlockchain__PrimaryText>
        <FormattedMessage
          defaultMessage="Lower cost"
          id="layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.lowercost"
          values={Object {}}
        />
      </ExchangeByBlockchain__PrimaryText>
      <ExchangeByBlockchain__DarkText
        hideOnMobile={true}
      >
         - 
      </ExchangeByBlockchain__DarkText>
      <FormattedMessage
        defaultMessage="Super competitive crypto exchange prices"
        id="layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.lowercost_description"
        values={Object {}}
      />
    </ExchangeByBlockchain__DarkText>
  </ExchangeByBlockchain__Row>
  <ExchangeByBlockchain__Row
    marginBottom="8px"
  >
    <ExchangeByBlockchain__DarkText>
      <ExchangeByBlockchain__PrimaryText>
        <FormattedMessage
          defaultMessage="Live rates"
          id="layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.liverates"
          values={Object {}}
        />
      </ExchangeByBlockchain__PrimaryText>
      <ExchangeByBlockchain__DarkText
        hideOnMobile={true}
      >
         - 
      </ExchangeByBlockchain__DarkText>
      <FormattedMessage
        defaultMessage="You always get the most up to date price"
        id="layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.liverates_Description"
        values={Object {}}
      />
    </ExchangeByBlockchain__DarkText>
  </ExchangeByBlockchain__Row>
  <ExchangeByBlockchain__Row
    marginBottom="32px"
  >
    <ExchangeByBlockchain__DarkText>
      <ExchangeByBlockchain__PrimaryText>
        <FormattedMessage
          defaultMessage="Higher limits"
          id="layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.higherlimits"
          values={Object {}}
        />
      </ExchangeByBlockchain__PrimaryText>
      <ExchangeByBlockchain__DarkText
        hideOnMobile={true}
      >
         - 
      </ExchangeByBlockchain__DarkText>
      <FormattedMessage
        defaultMessage="Limits from $1,000-$25,000"
        id="layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.higherlimits_description"
        values={Object {}}
      />
    </ExchangeByBlockchain__DarkText>
  </ExchangeByBlockchain__Row>
  <ExchangeByBlockchain__Row>
    <LinkContainer
      activeClassName="active"
      exact={false}
      replace={false}
      strict={false}
      to="/exchange"
    >
      <ExchangeByBlockchain__GetStartedButton>
        <FormattedMessage
          defaultMessage="Go To Exchange"
          id="layouts.wallet.trayright.whatsnew.whatsnewcontent.exchangebyblockchain.gotoexchange"
          values={Object {}}
        />
      </ExchangeByBlockchain__GetStartedButton>
    </LinkContainer>
  </ExchangeByBlockchain__Row>
</ExchangeByBlockchain__Container>
`;
